waku/store/v3/store.proto

```diff
@@ -0,0 +1,29 @@
+syntax = "proto3";
+
+// 13/WAKU2-STORE rfc: https://rfc.vac.dev/spec/13/
+// Protocol identifier: /vac/waku/store/3.0.0
+package waku.store.v3;
+
+import "waku/message/v1/message.proto";
+
+
+message HistoryRequest {
+  string request_id = 1;
+  optional string pubsub_topic = 10;
+  repeated string content_topic = 11;
+  optional int64 time_start = 12;
+  optional int64 time_end = 13;
+  // 50 Reserved
+  optional bytes pagination_cursor = 51;
+  optional bool pagination_forward = 52;
+  optional uint64 pagination_limit = 53;
+}
+
+message HistoryResponse {
+  string request_id = 1;
+  optional uint32 status_code = 10;
```

Comment:
Assuming this would include "success" codes similar to a `200 OK`? I'm generally OK (200 :P) with that, though linking [this](https://status-im.github.io/nim-style-guide/errors.status.html) general reasoning on mixing error codes with success codes. Any reason not to limit this to error codes? Since any number of codes could be considered "success" (even if we limit it to `2xx`), a client would probably have to rely on checking availability of messages in any case to determine if it was successful or use some slightly more complicated matching logic in case the client's success code list is outdated. If it receives an empty list with a success code that it does not recognise, it may wrongly determine that the query failed rather than just having returned an empty response. The error check becomes very simple if only included on error condition with an `errorCode.isSome()` check.

Reply:
That's a good point:

> If it receives an empty list with a success code that it does not recognise, it may wrongly determine that the query failed rather than just having returned an empty response.

In the success case, I am considering using the _not present_ state and the _default value (0)_ as the `200 OK` HTTP equivalent. So there is no confusion, and at the implementation, we can do the `statusCode.isNone()` or `statusCode != 0`.

- If the code is not present or it is the default value: OK
- If the code is present and my implementation supports the code: KO (known issue)

I am not sure about the backward compatibility case: _If the code is present, but my implementation does not understand the code_. Here we have two interpretation options:

- KO (Unsupported/Unknown status code)
- OK (Ignore the code)

IMO it depends on the use case to pick one or another strategy.

> Assuming this would include "success" codes similar to a `200 OK`? I'm generally OK (200 :P) with that, though linking [this](https://status-im.github.io/nim-style-guide/errors.status.html) general reasoning on mixing error codes with success codes. Any reason not to limit this to error codes?

When I think of status codes, I am not only thinking about errors like `4xx` and `5xx`, 
but also HTTP's `1xx` and `3xx` families of codes. For example, the 1xx family of codes is purely informational. In the future, we may need support responses with extra information (e.g., a "cursor format" info negotiation compatible with the Waku archive implementation).

Maybe something to consider is to rename the `status_desc` to something like `status_info`, `status_meta`, `status_data`, or `status_ext` since it can contain extra information extending the status code.